src/legacy/core_plugins/kibana/public/discover/components/histogram/histogram.tsx

```diff
@@ -0,0 +1,264 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiIcon, EuiSpacer } from '@elastic/eui';
+import moment from 'moment-timezone';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  AnnotationDomainTypes,
+  Axis,
+  Chart,
+  HistogramBarSeries,
+  getAnnotationId,
+  getAxisId,
+  getSpecId,
+  LineAnnotation,
+  Position,
+  ScaleType,
+  Settings,
+  RectAnnotation,
+  TooltipValue,
+} from '@elastic/charts';
+
+import { i18n } from '@kbn/i18n';
+
+import { getChartTheme } from 'ui/elastic_charts';
+import chrome from 'ui/chrome';
+// @ts-ignore: path dynamic for kibana
+import { timezoneProvider } from 'ui/vis/lib/timezone';
+
+export interface DiscoverHistogramProps {
+  chartData: any;
+  timefilterUpdateHandler: (ranges: { xaxis: { from: number; to: number } }) => void;
+}
+
+export class DiscoverHistogram extends Component<DiscoverHistogramProps> {
+  static propTypes = {
+    chartData: PropTypes.object,
+    timefilterUpdateHandler: PropTypes.func,
+  };
+
+  onBrushEnd = (min: number, max: number) => {
+    const range = {
+      xaxis: {
+        from: min,
+        to: max,
+      },
+    };
+
+    this.props.timefilterUpdateHandler(range);
+  };
+
+  onElementClick = (xInterval: number) => (elementData: any) => {
```

Comment:
This type annotation should be `(xInterval: number) => (elementData: GeometryValue[]) => { ... }` but `GeometryValue` wasn't being exported; resolved in this PR https://github.com/elastic/elastic-charts/pull/272 in the charts libary, will update the annotation once that's merged.

Reply:
resolved https://github.com/elastic/kibana/pull/36517/commits/4e3b2a52ba77d3a64cf9041d9fc33205599dc088